stdlib/lang-parser/parserModule/lexer.bal

```diff
@@ -0,0 +1,426 @@
+                }
+                //keeps track whether the string literal is incomplete
+                boolean incompleteStr = true;
+                //length of the string
+                int strPos = 0;
+
+                //run while we consume another inverted comma and complete the string literal
+                while (incompleteStr) {
+                    currChar = self.nextLexeme();
+                    self.position += 1;
+                    strPos += 1;
+                    //append the string with the current character
+                    str = str + currChar;
+                    if (self.buffer.lookAhead() == singleQuoteSym) {
+                        currChar = self.nextLexeme();
+                        self.position += 1;
+                        strPos += 1;
+                        str = str + currChar;
+                        incompleteStr = false;
+                        self.tokenIndex += 1;
+                        return {
+                            tokenType: QUOTED_STRING_LITERAL,
+                            text: str,
+                            startPos: self.position - strPos,
+                            endPos: self.position,
+                            lineNumber: self.lineNum,
+                            index: self.tokenIndex,
+                            whiteSpace: self.wpStack.getWhiteSpace()
+                        };
+                    }
+                    //loop until Eof
+                    if (self.buffer.lookAhead() == "") {
+                        incompleteStr = false;
+                        //if no inverted comma is found,all the characters until Eof is captured and returned as an Lexer Error Token
+                        return {
+                            tokenType: LEXER_ERROR_TOKEN,
+                            text: str,
+                            startPos: self.position - strPos,
+                            endPos: self.position,
+                            lineNumber: self.lineNum,
+                            index: self.tokenIndex,
+                            whiteSpace: self.wpStack.getWhiteSpace()
+                        };
+
+                    }
+                }
+            } else if (isDigit(currChar)) { // integers
+                string numb = currChar;
+                int digitPos = 0;
+                //keep track if the number is a valid number
+                boolean validNum = true;
+                //append the number while the lookahead character is also a number
+                while (isDigit(self.buffer.lookAhead())) {
+                    currChar = self.nextLexeme();
+                    self.position += 1;
+                    digitPos += 1;
+                    numb = numb + currChar;
+                }
+                //no identifier can be names with number to the front, if occured, it is returned as an lexer Error Token
+                if (isLetter(self.buffer.lookAhead())) {
+                    validNum = false;
```

Comment:
`validNum` isn't used after this point in this block right, before the return? Do we need this assignment?

Reply:
`validNum` is used in the next block to check if the token is a valid number and to return the `NUMBER` token

Reply:
Within this if block, we always return at L176 right? So the next block wouldn't be executed?

Reply:
Once the code execusion enters this `if` block, it won't reach the next `if` block since there is a return statement at the end of this `if` block.

Reply:
yes, I removed the assignment.